cooltools/cli/coverage.py

```diff
@@ -0,0 +1,116 @@
+from calendar import c
+import click
+import cooler
+from numpy import cov
+
+from .. import coverage
+
+from . import cli
+from .. import api
+from ..lib.common import make_cooler_view
+from ..lib.io import read_viewframe_from_file
+import bioframe
+import multiprocessing as mp
+
+
+@cli.command()
+@click.argument(
+    "cool_path", metavar="COOL_PATH", type=str, nargs=1,
+)
+@click.option(
+    "--output",
+    "-o",
+    help="Specify output file name to store the coverage in a tsv format.",
+    type=str,
+    required=False,
+)
+@click.option(
+    "--ignore-diags",
+    help="The number of diagonals to ignore. By default, equals"
+    " the number of diagonals ignored during IC balancing.",
+    type=int,
+    default=None,
+    show_default=True,
+)
+@click.option(
+    "--store",
+    help="Append columns with coverage (cis_raw_cov, tot_raw_cov) "
+    " to the cooler bin table.",
+    is_flag=True,
+)
+@click.option(
+    "--chunksize",
+    help="Split the contact matrix pixel records into equally sized chunks to"
+    " save memory and/or parallelize. Default is 10^7",
+    type=int,
+    default=1e7,
+    show_default=True,
+)
+@click.option(
+    "--bigwig",
+    help="Also save output as bigWig files for cis and total coverage"
+    " with the names <output>.<cis/tot>.bw",
+    is_flag=True,
+    default=False,
+)
+@click.option(
+    "-p",
+    "--nproc",
+    help="Number of processes to split the work between."
+    " [default: 1, i.e. no process pool]",
+    default=1,
+    type=int,
+)
+def coverage(
+    cool_path, output, ignore_diags, store, chunksize, bigwig, nproc,
+):
+    """
+    Calculate the sums of cis and genome-wide contacts (aka coverage aka marginals) for
+    a sparse Hi-C contact map in Cooler HDF5 format.
+    Note that the sum(tot_cov) from this function is two times the number of reads
+    contributing to the cooler, as each side contributes to the coverage.
+
+    COOL_PATH : The paths to a .cool file with a balanced Hi-C map.
+
+    """
+
+    clr = cooler.Cooler(cool_path)
+
+    if nproc > 1:
+        pool = mp.Pool(nproc)
+        _map = pool.imap
+    else:
+        _map = map
+
+    cis_cov, tot_cov = api.coverage.coverage(
+        clr, ignore_diags=ignore_diags, chunksize=chunksize, map=_map, store=store
+    )
+
+    if nproc > 1:
```

Comment:
https://github.com/open2c/cooltools/blob/f69981e346b662082ea3608d06bf217b1853check out this pattern 98a4/cooltools/api/expected.py#L1252
dunno why we do this

Reply:
https://github.com/open2c/cooltools/blob/f69981e346b662082ea3608d06bf217b185398a4/cooltools/api/expected.py#L1253